commercetools-models/src/test/java/io/sphere/sdk/carts/commands/CartUpdateCommandIntegrationTest.java

```diff
@@ -603,6 +606,76 @@ public void setLineItemTotalPrice() throws Exception {
     }
 
     @Test
+    public void setCartTotalTax() throws Exception {
+        withFilledCartWithTaxMode(client(), TaxMode.EXTERNAL_AMOUNT, cart -> {
+            final MonetaryAmount externalTotalGross = MoneyImpl.ofCents(1000, EUR);
+            final SetCartTotalTax setCartTotalTax = SetCartTotalTax.of(externalTotalGross);
+            final Cart updatedCart = client().executeBlocking(CartUpdateCommand.of(cart, setCartTotalTax));
+
+            assertThat(updatedCart.getTaxedPrice().getTotalGross()).isEqualTo(externalTotalGross);
+        });
+    }
+
+    @Test
+    public void setShippingMethodTaxAmount() throws Exception {
+        withShippingMethodForGermany(client(), shippingMethod -> {
+            withCustomLineItemFilledCartWithTaxMode(client(), TaxMode.EXTERNAL_AMOUNT, cart -> {
+                final Cart cartWithShippingMethod = client().executeBlocking(CartUpdateCommand.of(cart, SetShippingMethod.of(shippingMethod)));
+                final ExternalTaxRateDraft taxRate = ExternalTaxRateDraftBuilder
+                        .ofAmount(1.0, "Test Tax", CountryCode.DE)
+                        .build();
+                final MonetaryAmount totalGross = MoneyImpl.ofCents(1000, EUR);
+                final ExternalTaxAmountDraftDsl taxAmountDraft = ExternalTaxAmountDraftBuilder
+                        .of(totalGross, taxRate)
+                        .build();
+                final SetShippingMethodTaxAmount setShippingMethodTaxAmount = SetShippingMethodTaxAmount.of(taxAmountDraft);
+
+                final Cart updatedCart = client().executeBlocking(CartUpdateCommand.of(cartWithShippingMethod, setShippingMethodTaxAmount));
+
```

Comment:
Shouldn't there be an assertion checking that the shipping method was actually set, price and all?